experimental/oak_baremetal_channel/SPEC.MD

```diff
@@ -0,0 +1,121 @@
+
+The Trusted Runtime MUST NOT send Request Message.
+
+### Handling an invocation
+
+The Trusted Runtime MUST respond to each Request Message with a single Response
+Message. It SHOULD respond to Request Messages in first-in-first-out order, but
+MAY prioritize some requests. The Trusted Runtime MAY issue Response Messages
+before the Request Message has been received in its entirety. Sending a Response
+Messages concludes the invocation.
+
+The Untrusted Loader MUST NOT send Response Messages.
+
+## Messages
+
+A message forms one half of an Invocation.
+
+### Request Message Definition
+
+A byte encoded request message MUST consist of the following seqeuential fields:
+
+- length, u32, little endian: length of the entire message, including this field
+- invocation_id, u32, little endian: Identifies the invocation the request
+  initiates
+- method_id, u32, little endian: identifies the method that is being invoked
+- padding, 4 bytes: reserved to maintain 8 byte alignment. MUST be unset and
+  ignored
+- body, variable length byte array: the method’s arguments, encoded with
+  flatbuffer
+
+### Response Message Definition
+
+A byte encoded response message MUST consist of the following seqeuential
+fields:
+
+- length, u32, little endian: length of the entire message, including this field
+- invocation_id, u32, little endian: MUST match the invocation_id of the
+  relevant Request Message
+- status_code, u32, little endian: response status as a
+  [gRPC status code](https://grpc.github.io/grpc/core/md_doc_statuscodes.html)
+- status_message_length, u32, little endian: length of the status message
+- status_message, utf-8 encoded string: developer-facing error message
+- body, variable length byte array: the method’s return value, encoded with
+  flatbuffer
+
+### Sending Messages
+
+To send a message, it MUST first be encoded into frames. The resulting frame set
+MUST then be send as an uninterrupted sequence.
+
+### Receiving Messages
+
+To receive a message the recipent incrementaly reads frames from the
+communication channel. The recipent MAY start parsing a message before all of
+its frames have been received.
+
+### Encoding Messages into Frames
+
+The byte representation of a message MUST be chunked. The last chunk MAY be
+shorter than 63,936 bytes (the maximum size of a frame body). Any preceeding
+chunks MUST be exactly 63,936 bytes in length.
+
+A frame MUST then be formed for each resulting message chunk, with the message
+chunk forming the frame body.
+
+Messages shorter than 63,936 bytes MUST be encoded as a single frame.
+
+### Decoding Frames into Messages
+
+Messages are parsed from Frames by appending the frame bodies in the order they
+are received.
+
+## Frames
+
+Messages are sent and received as frames. Each message can be fragmented into
+one or more frames.
+
+### Frame Definition
+
+A byte encoded frame MUST consist of the following seqeuential fields:
+
+- padding, four bytes: for future use
```

Comment:
The idea would be to use this for a message_id in the future, allowing us to support multiplexing